lib/utils/display.js

```diff
@@ -109,142 +128,213 @@ class Display {
   constructor ({ stdout, stderr }) {
     this.#stdout = setBlocking(stdout)
     this.#stderr = setBlocking(stderr)
+
+    // Handlers are set immediately so they can buffer all events
     process.on('log', this.#logHandler)
+    process.on('output', this.#outputHandler)
   }
 
   off () {
     process.off('log', this.#logHandler)
-    this.#logBuffer.length = 0
+    this.#logState.buffer.length = 0
+
+    process.off('output', this.#outputHandler)
+    this.#outputState.buffer.length = 0
+
     if (this.#progress) {
       this.#progress.stop()
     }
   }
 
-  load ({ loglevel, chalk, timing, unicode, progress, json, heading }) {
-    this.#chalk = chalk
-    this.#colors = COLOR_PALETTE({ chalk })
+  load ({
+    loglevel,
+    stdoutColor,
+    stdoutChalk,
+    stderrColor,
+    stderrChalk,
+    timing,
+    unicode,
+    progress,
+    json,
+    heading,
+  }) {
+    this.#stdoutColor = stdoutColor
+    this.#stdoutChalk = stdoutChalk
+
+    this.#stderrColor = stderrColor
+    this.#stderrChalk = stderrChalk
+
+    this.#logColors = COLOR_PALETTE({ chalk: stderrChalk })
 
     this.#levelIndex = LEVEL_OPTIONS[loglevel].index
     this.#timing = timing
     this.#json = json
     this.#heading = heading
 
+    // In silent mode we remove all the handlers
     if (this.#levelIndex <= 0) {
       this.off()
-    } else {
-      log.resume()
-      if (progress) {
-        this.#startProgress({ unicode })
-      }
+      return
     }
-  }
 
-  logTiming (...args) {
-    this.#logHandler(LEVELS.timing, ...args)
-  }
+    // Emit resume event on the logs which will flush output
+    log.resume()
 
-  forceLog (level, ...args) {
-    // This will show the log regardless of the current loglevel, except when silent
-    this.#logHandler({ level, force: true }, ...args)
-  }
+    // TODO: this should be a proc-log method `proc-log.output.flush`?
+    this.#outputHandler(OUTPUT_LEVELS.flush)
 
-  output (...args) {
-    // TODO: make this respect silent option
-    this.#stdout.write(format(...args))
+    this.#startProgress({ progress, unicode })
   }
 
-  outputError (...args) {
-    this.#stderr.write(format(...args))
+  // STREAM WRITES
+
+  // Write formatted and (non-)colorized output to streams
+  #stdoutWrite (options, ...args) {
+    this.#stdout.write(formatWithOptions({ colors: this.#stdoutColor, ...options }, ...args))
   }
 
-  outputBuffer (item) {
-    this.#outputBuffer.push(item)
+  #stderrWrite (options, ...args) {
+    this.#stderr.write(formatWithOptions({ colors: this.#stderrColor, ...options }, ...args))
   }
 
-  flushOutput (jsonError) {
-    if (!jsonError && !this.#outputBuffer.length) {
+  // HANDLERS
+
+  // Public class field so it can be passed directly as a listener
```

Comment:
I don't understand this comment.

Reply:
It should be [`Private class field`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_properties#private_fields) not `Public`.

The syntax used here will make `this` stay bound to the class instance, so it can be passed directly to the `process.on` and `process.off` listeners.

For example:
```js
class MyClass {
  a = 1

  constructor() {
    process.on('log', this.#method)
    process.on('log', this.#field)
  }

  // Private method
  #method(...args) {
    console.log('method', this.a, args)
  }

  // Private class field
  #field = (...args) => {
    console.log('field', this.a, args)
  }
}

new MyClass()

process.emit('log', 'a')

// This will log
// method undefined [ 'a' ]
// field 1 [ 'a' ]
```

Reply:
// Arrow function assigned to a private class field so it can be passed directly as a listener and still reference `this`

I understand now.